### PR TITLE
Initial port to 1.0

### PR DESCRIPTION
**Pull request by ebidel**
R: @cdata @notwaldorf @crhym3

The new styling system has a few bugs in it. For now, I made the demo work under Shady and Shadow DOM with the duplicated `<style is="custom-style" include="sunburst-theme"></style> ` in the element and the main document (demo page).

### CODE REVIEW

File: highlightjs-element.html [hunk 1]
```diff
@@ -21,63 +24,62 @@
...
-        display: block;
-      }
-    </style>
-
-    <link href="../highlightjs/styles/default.css" rel="stylesheet" />
-    <link href="../highlightjs/styles/atelier-forest.light.css" rel="stylesheet" />
-
-    <pre id="content"></pre>
-
+    <style include="sunburst-theme"></style>
```
Comment (crhym3) [inline highlightjs-element.html]: What if `highlightjs-element` didn't include any styling and the host page did it instead? I'm thinking about custom styling.

File: highlightjs-element.html [hunk 2]
```diff
@@ -21,63 +24,62 @@
...
+    <style>:host { display: block; }</style>
+    <pre id="content" class="hljs"></pre>
```
Comment (crhym3) [inline highlightjs-element.html]: AFAIK `<pre><code>...</code></pre>` is recommended by HTML5

File: bower.json [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+    "code",
+    "syntax"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/Polymer/highlightjs-element.git"
+  },
   "dependencies": {
-    "polymer": "Polymer/polymer#master",
+    "polymer": "Polymer/polymer#^1.0.0",
```
Comment (crhym3) [inline bower.json]: Why not `1.1`?

File: highlightjs-import.html [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-    Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+    Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
     This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
     The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
     The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
     Code distributed by Google as part of the polymer project is also
     subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 -->
-<script src="../highlightjs/highlight.pack.js"></script>
+<script src="../highlightjs/highlight.pack.js"></script>
```
Comment (crhym3) [inline highlightjs-import.html]: Last time I checked this file was about 500Kb :(

File: highlightjs-element.html (hunk 1 above)
Comment (ebidel) [inline highlightjs-element.html]: Yes it really should, but out of the box the stylesheets won't work in the main document if you're using native shadow dom. They'd need to be rewritten or we need to come up with a technique to load them dynamically. There are a couple of bug around this (noted in the code) so I didn't both trying to make it work just yet.

File: highlightjs-element.html (hunk 2 above)
Comment (ebidel) [inline highlightjs-element.html]: Done

File: bower.json (hunk 3 above)
Comment (ebidel) [inline bower.json]: It will auto-upgrade. But I'll be explicit. Done

**Comment by crhym3** [comment]
LGTM :+1:

**Pull request merged**